lib/sqlalchemy/dialects/mysql/mariadbconnector.py

```diff
@@ -0,0 +1,182 @@
+# mysql/mariadb.py
+# Copyright (C) 2020 the SQLAlchemy authors and contributors
+# <see AUTHORS file>
+#
+# This module is part of SQLAlchemy and is released under
+# the MIT License: http://www.opensource.org/licenses/mit-license.php
+
+"""
+
+.. dialect:: mysql+mariadb
+    :name: MariaDB Connector/Python
+    :dbapi: mariadb
+    :connectstring: mysql+mariadb://<user>:<password>@<host>[:<port>]/<dbname>
+    :url: https://pypi.org/project/mariadb/
+
+Driver Status
+-------------
+
+MariaDB Connector/Python enables python programs to access MariaDB and MySQL
+databases, using an API which is compliant with the Python DB API 2.0 (PEP-249).
+It is written in C and uses MariaDB Connector/C client library for client server
+communication. 
+
+Status: stable
+
+.. mariadb: https://github.com/mariadb-corporation/mariadb-connector-python
+
+"""
+import os
+import re
+
+from .base import MySQLCompiler
+from .base import MySQLDialect
+from .base import MySQLExecutionContext
+from .base import MySQLIdentifierPreparer
+from .base import TEXT
+from ... import sql
+from ... import util
+from distutils.version import StrictVersion
+
+import mariadb.constants.CLIENT as CLIENT
+
+mariadb_cpy_minimum_version="1.0.1"
+
+class MySQLExecutionContext_mariadbconnector(MySQLExecutionContext):
+    pass
+
+class MySQLCompiler_mariadbconnector(MySQLCompiler):
+    pass
+
+class MySQLIdentifierPreparer_mariadbconnector(MySQLIdentifierPreparer):
+    pass
+
+# MariaDB binary protocol doesn't support XA yet, so we need
+# to rewrite the statement. See https://jira.mariadb.org/browse/MDEV-16708
+
+def check_unsupported_xa(statement, parameter):
+    if parameter is None or parameter.__len__() != 1:
+        return None
+    sql = re.sub(re.compile("/\\*.*?\\*/",re.DOTALL ) ,"" ,statement)
```

Comment:
**mike bayer** (zzzeek) wrote:

while we really don't need XA support in any case, this approach is too magical and slow, see next comment.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2166

Reply:
**Gord Thompson** (gordthompson) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2166